magicbot/magic_tunable.py

```diff
@@ -1,13 +1,23 @@
+import collections.abc
 import functools
 import inspect
+import typing
 import warnings
-from typing import Callable, Generic, Optional, TypeVar, overload
+from typing import Callable, Generic, Optional, Sequence, TypeVar, Union, overload
 
-from ntcore import NetworkTableInstance, Value
+import ntcore
+from ntcore import NetworkTableInstance
 from ntcore.types import ValueT
 
+
+class StructSerializable(typing.Protocol):
```

Comment:
This protocol probably shouldn't belong here...

Reply:
Probably should end up in wpimath, but I don't see why it can't be in two places.

Reply:
I presume you mean wpiutil? I'll follow up with a PR over there then.